lepton/azure.go

```diff
@@ -386,74 +409,70 @@ func (a *Azure) CreateInstance(ctx *Context) error {
 	c := ctx.config
 	bucket := c.CloudConfig.BucketName
 	if bucket == "" {
-		bucket = os.Getenv("AZURE_STORAGE_ACCOUNT")
+		bucket = a.storageAccount
 	}
 	if bucket == "" {
 		fmt.Println("AZURE_STORAGE_ACCOUNT should be set otherwise logs can not be retrieved.")
 		os.Exit(1)
 	}
 	location := a.getLocation(ctx.config)
 
-	debug := false
-
 	vmName := ctx.config.CloudConfig.ImageName + strconv.FormatInt(time.Now().Unix(), 10)
-	fmt.Printf("spinning up:\t%s\n", vmName)
+	ctx.logger.Log("spinning up:\t%s\n", vmName)
 
 	// create virtual network
+	ctx.logger.Info("creating virtual network with id %s\n", vmName)
 	vnet, err := a.CreateVirtualNetwork(context.TODO(), location, vmName)
 	if err != nil {
-		fmt.Println(err)
-	}
-
-	if debug {
-		fmt.Printf("%+v\n", vnet)
+		ctx.logger.Error("error creating virtual network: %v\n", err)
```

Comment:
The idea here is to display error details if debug flag is enabled and return the short version of the error to the caller.

We're introducing a lot of repetition this way. We could create an entity responsible for logging errors and create errors.

Reply:
+1

Reply:
https://github.com/nanovms/ops/issues/779